### DISCUSSION TITLE
trying to switch to presence mode or "and" combination in search?

### DISCUSSION THREAD

**Discussion by psifertex** [squidfunk/mkdocs-material#7356] [Q&A]
I'm eagerly waiting the [new search](https://github.com/squidfunk/mkdocs-material/issues/6307) but in the meantime, I'm curious if anyone has any work-arounds for the default lunrjs behavior of "or" keyword combining. Looking upstream I see that [adding a `+`](https://lunrjs.com/guides/searching.html#term-presence) for each keyword might work as a work-around?

There may also be some settings that can specified?

Not sure the best way to do either of those in the current mkdocs-material and I'm curious if anyone else has already had and solved the issue.

**Comment by squidfunk** [comment]
Thanks for asking. As far as I know from [studying the code base](https://github.com/olivernn/lunr.js/blob/master/lunr.js) while working on #6307, its not possible to change the "default presence" (i.e. optional to required) in lunr.js without either using the query builder, or prepending `+` to each and every term. You might fork this project and adjust the transform that we use to patch many of the oddities of lunr.js through [query transforms](https://github.com/squidfunk/mkdocs-material/blob/master/src/templates/assets/javascripts/integrations/search/query/transform/index.ts) (which have their limitations). The new search already supports this through configuration, so it should be worth the wait. I currently have a much bigger refactoring of this project on my plate that must preceed my work on search, but I will shortly return to finish the search once that's done. I'm confident to say that there will be a very, very big update this year.

**Reply by kamilkrzyskow** [reply]
To add to this, I remember another discussion here:
- https://github.com/squidfunk/mkdocs-material/discussions/6885#discussioncomment-8753778
where it turned out there is a QueryLexer with its own tokenizer, that is separate from the one overridden by the `mkdocs.yml` settings of the search plugin.

Also query transforms seem to be a good case for a bundle.js override, because they're unique searchable strings, I've provided an example of such override hook here:
- https://github.com/squidfunk/mkdocs-material/discussions/6844#discussioncomment-8670472

**Reply by squidfunk** [reply]
Thanks for digging this up! Always good to have more context. However, tokenization is one phase too late, as it runs after the lexer which converts the query string into an AST, so phrases have already been joined with OR (by marking them as optional).
